src/js/themes/base.d.ts

```diff
@@ -22,7 +33,20 @@ export declare const generate: (
   scale?: number,
 ) => DeepReadonly<ThemeType>;
 
-type ExtendType = string | ((...args: any) => void);
+type ExtendProps<TProps> = ThemedStyledProps<TProps, ThemeType>;
+type ExtendValue<TProps> = string | FlattenSimpleInterpolation | FlattenInterpolation<ExtendProps<TProps>>;
+type ExtendFn<TProps> = (props: ExtendProps<TProps>) => ExtendValue<TProps>;
```

Comment:
can we add a comment for each type and explain its purpose and value?

Reply:
Sounds good, I've added some comments. Hope I explained well enough, let me know if anything doesn't make sense.